.github/workflows/tests.yaml

```diff
@@ -8,66 +8,49 @@ env:
 on: [workflow_call]
 
 jobs:
-  jest:
+  install-test-env:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-      - name: Setup Node.js
-        uses: actions/setup-node@v1
-        with:
-          node-version: '16'
-
-      - name: Cache node modules
-        uses: actions/cache@v2
-        env:
-          cache-name: cache-node-modules
-        with:
-          path: ~/.npm
-          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
-          restore-keys: |
-            ${{ runner.os }}-build-${{ env.cache-name }}-
-            ${{ runner.os }}-build-
-            ${{ runner.os }}-
+      - name: dev env setup
+        uses: ./.github/actions/test-env-setup
 
-      - name: Install Dependencies
+  jest:
+    needs: [install-test-env]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: dev env setup
+        uses: ./.github/actions/test-env-setup
+      - run: yarn install --frozen-lockfile
```

Comment:
yarn install should only be done once, in your `install-test-env` job. When you then run the `test-env-setup` action, your packages will be added to the job by restoring the yarn cache

Reply:
Okay I removed `test-env-setup` action and am using `dev-env-setup` action we are using in the end to end tests, as well as do our yarn install in that action.